libs/types/src/pools.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Centrifuge GmbH (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_primitives::Moment;
+use codec::{Decode, Encode};
+use frame_support::{traits::Get, BoundedVec, RuntimeDebug};
+use orml_traits::Change;
+use scale_info::TypeInfo;
+
+use super::*;
+
+// #[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo)]
```

Comment:
Is there a reason why we have this block here commented out?

Reply:
Thanks, I forgot to point that out! I wanted feedback on the idea @mustermeiszer had on converting the `PoolChanges` from a struct to an enum. If we do this, where in the frontend/App/wherever do we have to adjust or tell people?